common/scala/src/main/scala/whisk/core/database/ArtifactStore.scala

```diff
@@ -28,6 +28,11 @@ import whisk.common.Logging
 import whisk.common.TransactionId
 import whisk.core.entity.DocInfo
 
+abstract class StaleParameter(val value: Option[String])
+case object StaleOk extends StaleParameter(Some("ok"))
+case object StaleUpdateAfter extends StaleParameter(Some("update_after"))
+case object NoStale extends StaleParameter(None)
```

Comment:
Awesome!!

One nit: You can wrap the implementations into a companion object like:

```scala
object StaleParameter {
  case object Ok extends StaleParameter(Some("ok"))
  case object UpdateAfter extends StaleParameter(Some("update_after"))
  case object None extends StaleParameter(None)
}
```

I think that makes scoping a bit easier and its best practice.